hub/core/transform/test_transform.py

```diff
@@ -982,3 +1004,11 @@ def test_transform_bug_link(local_ds, cat_path):
         for i in range(9):
             assert ds[i].abc.numpy().shape == (900, 900, 3)
             assert ds[i].abc.shape == (900, 900, 3)
+
+
+def test_tensor_dataset_memory_leak(local_ds):
+    local_ds.create_tensor("image", htype="image", sample_compression="png")
+    add_images().eval(list(range(5)), local_ds, scheduler="threaded")
```

Comment:
maybe have it run over a little more iterations? does this fail on main?

Reply:
yes, it fails on main